### PR TITLE
BI-10714 Validate using new criteria

### PR DESCRIPTION
**Pull request by mattch1**
Changed cirteria to validate company numbers that are alphanumeric and not over 10 characters long as the rpevious valaidation was to narrow.

Removed length check as the regex does this for us, altered error message slightly to allow for this.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/validation/CompanyNumberValidationInterceptor.java [hunk 1]
```diff
@@ -22,9 +22,6 @@
     @Value("${MAX_ID_LENGTH}")
     private int maxIdLength;
 
-    @Value("${MAX_COMPANY_NUMBER_LENGTH}")
```
Comment (mwestacott) [inline src/main/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/validation/CompanyNumberValidationInterceptor.java:25]: Wondering why this has gone and why we use a MAX_ID_LENGTH here? I'd have thought a company number max length is quite different to an id max length.

Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/validation/CompanyNumberValidationInterceptor.java:25]: It is. This is being used to truncate the string as we do not want to be logging very long strings.

File: src/test/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/validation/CompanyNumberValidationInterceptorTest.java [hunk 2]
```diff
@@ -24,16 +24,15 @@
 class CompanyNumberValidationInterceptorTest {
 
     static Stream<String> validStrings() {
-        return Stream.of("11111111", "A1111111", "AB111111", "ab111111");
+        return Stream.of("11111111", "A1111111", "AB111111", "ab111111", "IP00366C", "1010101010");
```
[the remark this answers is not in the document]
Comment (mwestacott) [reply src/test/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/validation/CompanyNumberValidationInterceptorTest.java:27]: Makes sense. Would suggest adding an 11 character string to the invalidStrings() method below then, just to check the other side of the boundary - '10101010101'.

File: src/main/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/validation/CompanyNumberValidationInterceptor.java (hunk 1 above)
Comment (mwestacott) [reply src/main/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/validation/CompanyNumberValidationInterceptor.java:25]: So sounds like we should not use / reference MAX_ID_LENGTH here but keep the MAX_COMPANY_NUMBER_LENGTH, use this and set to 10 in the config (to match the max length in the reg exp).

File: src/test/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/validation/CompanyNumberValidationInterceptorTest.java [hunk 3]
```diff
@@ -44,8 +43,7 @@
     @BeforeEach
     void setEnvironment() {
         ReflectionTestUtils.setField(companyNumberValidationInterceptor, "maxIdLength", 50);
-        ReflectionTestUtils.setField(companyNumberValidationInterceptor, "maxCompanyNumberLength", 8);
-        ReflectionTestUtils.setField(companyNumberValidationInterceptor, "companyNumberPattern", "^([a-z]|[a-z][a-z])?\\d{6,8}$");
+        ReflectionTestUtils.setField(companyNumberValidationInterceptor, "companyNumberPattern", "^[A-Za-z0-9]{0,10}$");
```
Comment (mwestacott) [inline src/test/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/validation/CompanyNumberValidationInterceptorTest.java:46]: I think if the reg exp has changed we'll need to change this in the web layer too. See company.number.validator.ts. Please test with the web app before changing to confirm...

File: src/main/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/validation/CompanyNumberValidationInterceptor.java (hunk 1 above)
Comment (mwestacott) [reply src/main/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/validation/CompanyNumberValidationInterceptor.java:25]: Ok. Discussed this and now understand the motivation. Agreed to rename the 'maxIdLength' var though to something more helpful, like 'truncationLength'.

File: src/test/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/validation/CompanyNumberValidationInterceptorTest.java (hunk 3 above)
Comment (mattch1) [reply src/test/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/validation/CompanyNumberValidationInterceptorTest.java:46]: This work should be done on another ticket as we'll be changing and testing the web app and we want some visibility that we are doing this, so should not affect this PR.

Comment (mwestacott) [reply src/test/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/validation/CompanyNumberValidationInterceptorTest.java:46]: No shouldn't affect this PR but surely the work is related to the defect so we need another PR for the same JIRA ticket that fixes the web layer too.

**Review by mwestacott — APPROVED** [review]

File: src/test/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/validation/CompanyNumberValidationInterceptorTest.java (hunk 3 above)
Comment (mattch1) [reply src/test/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/validation/CompanyNumberValidationInterceptorTest.java:46]: We have a ticket that is linked to this JIRA ticket.

**Review by mwestacott — APPROVED** [review]

**Pull request merged**